Core.ApplicationServices/ItSystemUsageMigration/ItSystemUsageMigrationService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Core.ApplicationServices.Authorization;
+using Core.DomainModel;
+using Core.DomainModel.ItSystem;
+using Core.DomainServices;
+using Core.DomainServices.Extensions;
+
+namespace Core.ApplicationServices.ItSystemUsageMigration
+{
+    public class ItSystemUsageMigrationService : IItSystemUsageMigrationService
+    {
+        private IAuthorizationContext _authorizationContext;
+        private readonly IGenericRepository<ItSystem> _itSystemRepository;
+
+        public ItSystemUsageMigrationService(IAuthorizationContext authorizationContext, IGenericRepository<ItSystem> itSystemRepository)
+        {
+            _authorizationContext = authorizationContext;
+            _itSystemRepository = itSystemRepository;
+        }
+
+        public Return<IEnumerable<ItSystem>> GetUnusedItSystemsByOrganization(int organizationId, string nameContent, int limit)
+        {
+
+            var itSystems = _itSystemRepository
+                .AsQueryable()
+                .ByPublicAccessOrOrganizationId(organizationId)
+                .ExceptByInUsage(organizationId)
```

Comment:
Jeg er lidt utryg ved den konstruktion. Jeg vil hellere have en ExceptEntitiesWithIds som tager et array af Ider og laver en x=>ids.Contains(x) == false expression.

Det betyder at du først skal spørge efter all itsystemusages i din organisation og kalde Select(x=>x.id).ToList() og smide det med som input til din ExpectEntitiesWithId

Fordelen ved denne løsning er at du kun laver een query for at få dit konstante udfaldsrum med til "not in" query konstruktionen og du skal derfor ikke lave en sub query (i tabellen) for hvert resultat i supersettet